src/ai/backend/manager/services/users/actions/modify_user.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from typing import Any, Optional, override
+
+import sqlalchemy as sa
+
+from ai.backend.common.types import Sentinel
+from ai.backend.manager.actions.action import BaseActionResult
+from ai.backend.manager.models.user import UserRole, UserStatus
+from ai.backend.manager.services.users.actions.base import UserAction
+from ai.backend.manager.services.users.type import UserData
+
+
+@dataclass
+class ModifyUserAction(UserAction):
+    email: str
+    username: str | Sentinel = Sentinel.TOKEN
+    password: str | Sentinel = Sentinel.TOKEN
+    need_password_change: bool | Sentinel = Sentinel.TOKEN
+    full_name: str | Sentinel = Sentinel.TOKEN
+    description: str | Sentinel = Sentinel.TOKEN
+    is_active: bool | Sentinel = Sentinel.TOKEN
+    status: UserStatus | Sentinel = Sentinel.TOKEN
+    domain_name: str | Sentinel = Sentinel.TOKEN
+    role: UserRole | Sentinel = Sentinel.TOKEN
+    group_ids: list[str] | Sentinel = Sentinel.TOKEN
+    allowed_client_ip: Optional[list[str]] | Sentinel = Sentinel.TOKEN
+    totp_activated: bool | Sentinel = False
+    resource_policy: str | Sentinel = Sentinel.TOKEN
+    sudo_session_enabled: bool | Sentinel = False
+    main_access_key: Optional[str] | Sentinel = Sentinel.TOKEN
+    container_uid: Optional[int] | Sentinel = Sentinel.TOKEN
+    container_main_gid: Optional[int] | Sentinel = Sentinel.TOKEN
+    container_gids: Optional[list[int]] | Sentinel = Sentinel.TOKEN
```

Comment:
Should we represent the tri-state using `SENTINEL`?
I've used `TriState` in the `ModifyImage` mutation for representing tri-state.

https://github.com/lablup/backend.ai/pull/3997/files#diff-cc78343bb8958d0dce042be90206317b33ce11b04e1a667f37de562ab1010356R30

It would be best to standardize on one of the two approaches.
cc @HyeockJinKim